src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -2510,6 +2511,71 @@ public void testSignalWithStart() throws Exception {
         workflowClient.newUntypedWorkflowStub(execution, Optional.empty()).getResult(String.class));
   }
 
+  public static class TestConsistentQueryImpl implements QueryableWorkflow {
+    String state = "initial";
+
+    @Override
+    public String execute() {
+      TestActivities testActivities =
+          Workflow.newActivityStub(TestActivities.class, newActivityOptions2());
+
+      while (!state.equals("exit")) {
+        String oldState = state;
+        Workflow.await(() -> !Objects.equals(state, oldState));
+        try {
+          // Simulate longer workflow.
+          // Queries with EVENTUAL consistency will return stale state before this finishes.
+          Thread.sleep(1000);
```

Comment:
For reviewers, is there a better way to temporarily prolong workflow execution, so that is appears in-flight. This is already causing test flakiness.